### PR TITLE
Add setCommunicationsDevice menu

### PR DESCRIPTION
**Pull request by philburk**
Also move setSpeakerphoneOn into test activities.

### CODE REVIEW

File: apps/OboeTester/app/src/main/java/com/mobileer/audio_device/CommunicationDeviceSpinner.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        // devices which are currently connected.
+        mAudioManager.registerAudioDeviceCallback(new AudioDeviceCallback() {
+            @Override
+            public void onAudioDevicesAdded(AudioDeviceInfo[] addedDevices) {
+                updateDeviceList();
+            }
+
+            public void onAudioDevicesRemoved(AudioDeviceInfo[] removedDevices) {
+                updateDeviceList();
+                setSelection(0);
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/java/com/mobileer/audio_device/CommunicationDeviceSpinner.java:107]: updateDeviceList already this to the 0th device. Should we do this again?

File: apps/OboeTester/app/src/main/java/com/mobileer/audio_device/CommunicationDeviceSpinner.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                mDeviceAdapter.clear();
+                mDeviceAdapter.add(new AudioDeviceListEntry(CLEAR_DEVICE_ID,
+                        mContext.getString(R.string.clear)));
+                setSelection(0);
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.S) {
+                    List<AudioDeviceInfo> commDeviceList = mAudioManager.getAvailableCommunicationDevices();
+                    mCommDeviceArray = commDeviceList.toArray(new AudioDeviceInfo[0]);
+                    List<AudioDeviceListEntry> deviceList =
+                            AudioDeviceListEntry.createListFrom(
+                                    mCommDeviceArray, AudioManager.GET_DEVICES_OUTPUTS);
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/java/com/mobileer/audio_device/CommunicationDeviceSpinner.java:120]: What about input devices? Do we assume that communication devices have both input/output?

File: apps/OboeTester/app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -182,25 +182,13 @@
...
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:orientation="horizontal"
-        app:layout_constraintStart_toStartOf="@+id/setSpeakerphoneOn"
-        app:layout_constraintTop_toBottomOf="@+id/setSpeakerphoneOn"
+        app:layout_constraintStart_toStartOf="@+id/callbackSize"
+        app:layout_constraintTop_toBottomOf="@+id/callbackSize"
         >
         <CheckBox
             android:id="@+id/setBluetoothScoOn"
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/res/layout/activity_main.xml:194]: Should we move setBluetoothScoOn into CommunicationDeviceView too? setCommunicationDevice is meant to replace it as well.

File: apps/OboeTester/app/src/main/java/com/mobileer/oboetester/CommunicationDeviceView.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        mAudioManager.setSpeakerphoneOn(enabled);
+    }
+
+    private void showCommDeviceStatus() {
+        boolean enabled = mAudioManager.isSpeakerphoneOn();
+        String text = (enabled ? "ON" : "OFF");
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.S) {
+            AudioDeviceInfo commDeviceInfo = mAudioManager.getCommunicationDevice();
+            if (commDeviceInfo != null) {
+                text += ", CD#" + commDeviceInfo.getId();
```
Comment (robertwu1) [inline apps/OboeTester/app/src/main/java/com/mobileer/oboetester/CommunicationDeviceView.java:129]: nit: CD seems quite vague. Can we use a longer name?

Comment (philburk) [reply apps/OboeTester/app/src/main/java/com/mobileer/oboetester/CommunicationDeviceView.java:129]: I changed it to "CommDev=". More clear but still fits on the small screen.

File: apps/OboeTester/app/src/main/java/com/mobileer/audio_device/CommunicationDeviceSpinner.java (hunk 2 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/java/com/mobileer/audio_device/CommunicationDeviceSpinner.java:120]: The Java docs are not very clear but CommunicationsDevices are always OUTPUT. The input is chosen based on the output device. I added a comment.

File: apps/OboeTester/app/src/main/java/com/mobileer/audio_device/CommunicationDeviceSpinner.java (hunk 1 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/java/com/mobileer/audio_device/CommunicationDeviceSpinner.java:107]: Thanks. I removed it.

File: apps/OboeTester/app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (robertwu1) [reply apps/OboeTester/app/src/main/res/layout/activity_main.xml:194]: Yes, it was made to deprecate startBluetoothSco and stopBluetoothSco

**Review by robertwu1 — APPROVED** [review]

File: apps/OboeTester/app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (philburk) [reply apps/OboeTester/app/src/main/res/layout/activity_main.xml:194]: This doc mentions startBluetoothSco() but does not mention whether it replaces setBluetoothScoOn().
https://developer.android.com/reference/android/media/AudioManager#setCommunicationDevice(android.media.AudioDeviceInfo)

Comment (philburk) [reply apps/OboeTester/app/src/main/res/layout/activity_main.xml:194]: #1671 to track remaining SCO work

**Pull request merged**